src/librustdoc/html/render.rs

```diff
@@ -3547,6 +3547,10 @@ fn small_url_encode(s: &str) -> String {
      .replace("]", "%5D")
 }
 
+fn small_html_encode(s: &str) -> String {
```

Comment:
Why aren't you using `Escape`? `&`s appear in type parameters for impls so need to be escaped which `Escape` will handle but this doens't.

Reply:
I didn't understand the second part of your comment (and to answer the first, I didn't know it existed, I'll update!).

Reply:
[`impl<'a> Add<&'a i32> for i32`](https://doc.rust-lang.org/nightly/std/primitive.i32.html#impl-Add%3C%26%27a%20i32%3E) is a valid thing to attempt to print, so the `&` in there should be escaped properly in the html.